Packs/CheckPointHarmonyEndpoint/Integrations/CheckPointHarmonyEndpoint/CheckPointHarmonyEndpoint.py

```diff
@@ -0,0 +1,2703 @@
+
+
+def validate_filter_arguments(column_name: str | None = None, filter_type: str = None):
+    """Validate filter arguments values are allowed.
+
+    Args:
+        column_name (str, optional): The column name to filter by. Defaults to None.
+        filter_type (str, optional): The filter operator. Defaults to None.
+
+    Raises:
+        ValueError: Raise error in case column_name or filter_type values are not allowed.
+    """
+    if column_name and column_name not in COLUMN_NAMES:
+        raise ValueError(f"'column_name' must be one of the followings: {COLUMN_NAMES}.")
+
+    if filter_type and filter_type not in FILTER_TYPES:
+        raise ValueError(f"'filter_type' must be one of the followings: {FILTER_TYPES}.")
+
+
+def extract_query_filter(args: dict[str, Any]) -> list[dict[str, Any]]:
+    """Extract query filters from the specified arguments.
+
+    Args:
+        args (dict[str, Any]): Command arguments from XSOAR.
+
+    Returns:
+        list[dict[str, Any]]: The updated query filter according to Harmony requirements.
+    """
+    query_filter = []
+
+    if filter_by_query := args.get("filter"):
+        queries = argToList(filter_by_query, "' , ")
+
+        for query in queries:
+            query_parts = query.split(" ")
+            column_name = query_parts[0]
+            filter_type = query_parts[1]
+            filter_values = query_parts[2].replace("'", "")
+
+            validate_filter_arguments(column_name, filter_type)
+
+            query_filter.append(
+                {
+                    "columnName": column_name,
+                    "filterValues": argToList(filter_values),
+                    "filterType": filter_type,
+                }
+            )
+    for key, value in args.items():
+        if key in COLUMN_NAMES_MAPPER:
+            query_filter.append(
+                {
+                    "columnName": COLUMN_NAMES_MAPPER[key],
+                    "filterValues": argToList(value),
+                    "filterType": DEFAULT_FILTER_TYPE,
+                }
+            )
+
+    if computer_last_connection := args.get("computer_last_connection"):
+        computer_last_connection_times = argToList(computer_last_connection)
+        query_filter += [
+            {
+                "columnName": "computerLastConnection",
+                "filterValues": computer_last_connection_times[0],
+                "filterType": "Grater",
+            },
+            {
+                "columnName": "computerLastConnection",
+                "filterValues": computer_last_connection_times[1],
+                "filterType": "Smaller",
+            },
+        ]
+
+    if not query_filter:
+        raise DemistoException(
+            '''At least one of the following query arguments are required: computer_ids, computer_names, computer_ips,
+            computer_group_names, computer_types, computer_deployment_status, computer_last_connection, or filter.'''
```

Comment:
Can we use `COLUMN_NAMES_MAPPER.keys()` instead of stating the names explicitly? In case we will want to add another value in the future.

Reply:
COLUMN_NAMES_MAPPER.keys() doesn't cover all query arguments.